ColorItem doesn't accept 'h,s,b' string

Whilst writing tests for #332, I came across this issue:

From Karaf console:
```
openhab>send ColorItem1 1,2,3
Command has been sent successfully.
openhab> status ColorItem1
1,2,3
openhab> send ColorItem1 "1,4,3"
Command has been sent successfully.
openhab> status ColorItem1
1,4,3
```

I would expect ColorItem to accept a csv string but it didn't seem to:

```ruby
ColorItem1 << '1,2,3'
```
Would cause:
```
08:46:23.397 [ERROR] [ript.internal.ScriptEngineManagerImpl] - Error during evaluation of script 'file:/openhab/conf/automation/jsr223/ruby/personal/test.rb': org.jruby.embed.EvalFailedException: (NameError) no constructor for arguments (java.math.BigDecimal) on Java::OrgOpenhabCoreLibraryTypes::HSBType
```

--- a/features/color_item.feature
+++ b/features/color_item.feature
@@ -4,8 +4,8 @@ Feature: color_item
    Background:
      Given Clean OpenHAB with latest Ruby Libraries
      And items:
-       | type  | name  |
-       | Color | Color |
+       | type  | name  | state |
+       | Color | Color | 0,0,0 |
 
    Scenario Outline:  Color items can be updated
      Given code in a rules file:
@@ -18,3 +18,5 @@ Feature: color_item
        | command                     | final     |
        | HSBType.new(0, 100, 100)    | 0,100,100 |
        | HSBType.from_rgb(255, 0, 0) | 0,100,100 |
+       | '0,100,100'                 | 0,100,100 |
+       | '#FF0000'                   | 0,100,100 |

--- a/lib/openhab/dsl/types/hsb_type.rb
+++ b/lib/openhab/dsl/types/hsb_type.rb
@@ -1,5 +1,6 @@
 # frozen_string_literal: true
 
+require 'java'
 require_relative 'percent_type'
 
 module OpenHAB
@@ -65,6 +66,10 @@ module OpenHAB
               end
             end
 
+            # Convert strings using java class
+            return value_of(args.first) if args.length == 1 && args.first.is_a?(String)
+
+            # use super constructor for empty args
             return super unless args.length == 3
 
             # convert from several numeric-like types to the exact types

